services/crypto_transfer.proto

```diff
@@ -29,26 +37,60 @@ option java_multiple_files = true;
 import "basic_types.proto";
 
 /**
- * Transfers cryptocurrency among two or more accounts by making the desired adjustments to their
- * balances. Each transfer list can specify up to 10 adjustments. Each negative amount is withdrawn
- * from the corresponding account (a sender), and each positive one is added to the corresponding
- * account (a receiver). The amounts list must sum to zero. Each amount is a number of tinybars
- * (there are 100,000,000 tinybars in one hbar).  If any sender account fails to have sufficient
- * hbars, then the entire transaction fails, and none of those transfers occur, though the
- * transaction fee is still charged. This transaction must be signed by the keys for all the sending
- * accounts, and for any receiving accounts that have receiverSigRequired == true. The signatures
- * are in the same order as the accounts, skipping those accounts that don't need a signature. 
+ * Transfer HBAR and/or other tokens among two or more accounts and/or smart
+ * contracts.
+ *
+ * Transfers of HBAR or fungible/common tokens in this transaction are
+ * structured as a "double-entry" transfer list which debits one or more
+ * accounts, and separately credits one or more accounts. Each such transfer
+ * list may specify up to 10 individual credits or debits.<br/>
+ * Transfers of non-fungible/unique tokens in this transaction are
+ * structured as a "single-entry" transfer list, which both debits one account
+ * and credits another account in a single entry.
+ *
+ * At least one transfer MUST be present, this MAY be an HBAR transfer in
+ * `transfers`, or MAY be a token transfer in `tokenTransfers`.<br/>
+ * Either `transfers` or `tokenTransfers` MAY be unset, provided the other
+ * is set and not empty.<br/>
+ * If any one account with a debit in any transfer list holds insufficient
+ * balance to complete the transfer, the entire transaction SHALL fail, and
+ * all transfers SHALL NOT be completed.<br/>
+ * If any one account that is _sending_ an individual non-fungible/unique (NFT)
+ * token does not currently hold that unique NFT, the entire transaction SHALL
+ * FAIL, and all transfers SHALL NOT be completed.
+ * The transaction fee SHALL be charged for a transaction that fails due to
+ * insufficient balance or not holding the NFT to be transferred.<br/>
+ * Each account with any debit amounts in any transfer list MUST sign this
+ * transaction.<br/>
+ * Each account with any credit amounts in any transfer list that also has the
+ * `receiverSigRequired` flag set MUST sign this transaction.
+ *
+ * ### Block Stream Effects
+ * All debits and credits completed by this transaction SHALL be included in
+ * the transaction record transfer list.<br/>
```

Comment:
Do you mean StateChanges here ?

Reply:
We still have the transfer list in the `TransactionResult` message, and this refers to that transfer list.
StateChanges will transmit each credit or debit separately as account updates, not consolidated.